state/bakerystorage/storage.go

```diff
@@ -4,28 +4,32 @@
 package bakerystorage
 
 import (
+	"encoding/json"
 	"time"
 
 	"github.com/juju/errors"
-	"gopkg.in/macaroon-bakery.v1/bakery"
+	"gopkg.in/macaroon-bakery.v2-unstable/bakery"
+	"gopkg.in/macaroon-bakery.v2-unstable/bakery/mgostorage"
 	"gopkg.in/mgo.v2"
 )
 
-var expiryTimeIndex = mgo.Index{
-	Key:    []string{"expire-at"},
-	Sparse: true,
-
-	// We expire records when the clock time is one
-	// second older than the record's expire-at field
-	// value. It has to be at least one second, because
-	// mgo uses "omitempty" for this field.
-	ExpireAfter: time.Second,
+// MongoIndexes returns the indexes to apply to the MongoDB collection.
+func MongoIndexes() []mgo.Index {
```

Comment:
Don't we want to keep the existing expire-at index also so legacy macaroons will get deleted when required

Reply:
legacy macaroons will exist in a collection setup by a previous version of juju, so wouldn't those indices remain? Im pretty sure they aren't removed by this change, and there is no need for it to exist from this point onwards as we're just using new Juju.